specification/chaos/resource-manager/Microsoft.Chaos/preview/2024-12-01-preview/types/faultSimulations.json

```diff
@@ -0,0 +1,150 @@
+          },
+          "endTime": {
+            "type": "string",
+            "format": "date-time",
+            "description": "The time this fault simulation completed"
+          },
+          "expirationTime": {
+            "type": "string",
+            "format": "date-time",
+            "description": "The absolute expiration time after which this fault should be abandoned or cancelled"
+          },
+          "parameters": {
+            "$ref": "#/definitions/FaultSimulationParameters",
+            "description": "The parameters used when starting this fault simulation"
+          },
+          "description": {
+            "type": "string",
+            "description": "A description of the fault-simulation"
+          }
+        },
+        "description": "The fault simulation definition."
+      },
+      "FaultSimulationListResult": {
+        "type": "object",
+        "properties": {
+          "value": {
+            "type": "array",
+            "description": "Array of fault simulation resources",
+            "items": {
+              "$ref": "#/definitions/FaultSimulation"
+            }
+          },
+          "nextLink": {
+            "type": "string",
+            "description": "The URL to use for getting the next set of results."
+          }
+        },
+        "description": "List of fault simulation results associated with this resource"
+      },
+      "FaultSimulationParameters": {
+        "type": "object",
+        "required": [
+          "faultKind"
+        ],
+        "properties": {
+          "faultKind": {
+            "$ref": "#/definitions/FaultKind"
+          },
+          "constraints": {
+            "type": "object",
+            "description": "TODO_TODO_TODO",
+            "properties": {
+              "expirationTime": {
+                "type": "string",
+                "format": "date-time",
+                "description": "The absolute expiration timestamp (UTC) after which this fault simulation should be abandoned or reverted."
+              },
+              "maxDurationMinutes": {
+                "type": "integer",
+                "description": "The targeted duration that this simulation should run for."
+              }
+            },
+            "required": [
+              "expirationTime"
+            ]
+          }
+        },
+        "discriminator": "faultKind"
+      },
+      "ZoneDownFaultSimulationParameters":{
+        "type": "object",
+        "x-ms-discriminator-value": "ZoneDown",
+        "allOf": [
+          {
+            "$ref": "#/definitions/FaultSimulationParameters"
+          }
+        ],
+        "properties": {
+            "zones" : {
+              "type": "array",
+              "description": "One or more availability zone that an outage will be simulated on. Some providers may only support simulations on one zone at a time.",
+              "items":{
+                "type": "string",
+                "enum": [
```

Comment:
if you model this as an enum I believe ARM requires to add x-ms-enum. something like:

"x-ms-enum": {
              "name": "zones",
              "modelAsString": true,
              "values": [
                {
                  "value": "1",
                  "description": "Availability Zone 1."
                },
                {
                  "value": "2",
                  "description": "Availability Zone 2."
                },
                {
                  "value": "3",
                  "description": "Availability Zone 3."
                },
                {
                  "value": "4",
                  "description": "Availability Zone 4."
                },
                {
                  "value": "5",
                  "description": "Availability Zone 5."
                }
              ]
            }